### PR TITLE
Fb Issue6080: Add link to username in grids

### PR DESCRIPTION
**Pull request by ilikepies101**
If the logged in user has the permissions to view a user, the usernames displayed in a grid will have a link to the user details page.

### CODE REVIEW

File: api/src/org/labkey/api/query/UserIdRenderer.java [hunk 1]
```diff
@@ -64,4 +68,18 @@ public String getFormattedValue(RenderContext ctx)
...
         return super.getFormattedValue(ctx);
     }
+
+    @Override
+    public String renderURL(RenderContext ctx)
+    {
+        var container = ctx.getContainer();
+        var loggedInUser = ctx.getViewContext().getUser();
+        if(SecurityManager.canSeeUserDetails(container, loggedInUser))
+        {
```
Comment (labkey-klum) [inline api/src/org/labkey/api/query/UserIdRenderer.java]: styling convention : can we add a space between if and open paren?

File: api/src/org/labkey/api/query/UserIdRenderer.java [hunk 2]
```diff
@@ -64,4 +68,18 @@ public String getFormattedValue(RenderContext ctx)
...
+            int displayedUserId = (int)getBoundColumn().getValue(ctx);
+            ActionURL userURL = PageFlowUtil.urlProvider(UserUrls.class).getUserDetailsURL(container, displayedUserId, null);
```
Comment (labkey-klum) [inline api/src/org/labkey/api/query/UserIdRenderer.java]: you probably need to protect against an auto-unbox NPE here, suggestion would be to do something like:

Integer displayUserId = (Integer)getBoundColumn().getValue(ctx);
if (displayUserId != null)
{
   ....

File: api/src/org/labkey/api/query/UserIdRenderer.java [hunk 3]
```diff
@@ -64,4 +68,20 @@ public String getFormattedValue(RenderContext ctx)
...
+    @Override
+    public String renderURL(RenderContext ctx)
+    {
+        var container = ctx.getContainer();
+        var loggedInUser = ctx.getViewContext().getUser();
+        if (SecurityManager.canSeeUserDetails(container, loggedInUser))
+        {
+            Integer displayedUserId = (Integer)getBoundColumn().getValue(ctx);
+            if (displayedUserId != null){
+                ActionURL userURL = PageFlowUtil.urlProvider(UserUrls.class).getUserDetailsURL(container, displayedUserId, null);
```
Comment (labkey-klum) [inline api/src/org/labkey/api/query/UserIdRenderer.java]: sorry, can we get the brace on the next line, we usually use this style for javascript but not for java (I know it seems arbitrary), but when in doubt just be consistent with the rest of the code in the file.

Comment (ilikepies101) [reply api/src/org/labkey/api/query/UserIdRenderer.java]: Yup! Thanks for catching that it slipped by.

**Pull request merged**